functions/consumer/cassandra-consumer/src/main/java/org/springframework/cloud/fn/consumer/cassandra/CassandraConsumerConfiguration.java

```diff
@@ -82,19 +88,19 @@ public IntegrationFlow cassandraConsumerFlow(MessageHandler cassandraSinkMessage
 
 	@Bean
 	public MessageHandler cassandraSinkMessageHandler(ReactiveCassandraOperations cassandraOperations) {
-		CassandraMessageHandler cassandraMessageHandler =
-				this.cassandraSinkProperties.getQueryType() != null
-						? new CassandraMessageHandler(cassandraOperations, this.cassandraSinkProperties.getQueryType())
-						: new CassandraMessageHandler(cassandraOperations);
+		CassandraMessageHandler.Type queryType =
+				Optional.ofNullable(this.cassandraSinkProperties.getQueryType())
+						.orElse(CassandraMessageHandler.Type.INSERT);
+
+		CassandraMessageHandler cassandraMessageHandler = new CassandraMessageHandler(cassandraOperations, queryType);
 		cassandraMessageHandler.setProducesReply(true);
-		cassandraMessageHandler.setAsync(true);
-		if (this.cassandraSinkProperties.getConsistencyLevel() != null
-				|| this.cassandraSinkProperties.getTtl() > 0) {
+		int ttl = this.cassandraSinkProperties.getTtl();
+		ConsistencyLevel consistencyLevel = this.cassandraSinkProperties.getConsistencyLevel();
+		if (consistencyLevel != null || ttl > 0) {
 
 			WriteOptions.WriteOptionsBuilder writeOptionsBuilder = WriteOptions.builder();
```

Comment:
This is a great candidate for the enhanced switch statement - it reduces it down to this:
```java
	WriteOptions.WriteOptionsBuilder writeOptionsBuilder = switch (queryType) {
		case INSERT -> InsertOptions.builder();
		case UPDATE -> UpdateOptions.builder();
		default -> WriteOptions.builder();
	};
```

Reply:
Yeah, thought about that. But looks like I have mixed it with a pattern matching expression which is still preview: https://docs.oracle.com/en/java/javase/17/language/pattern-matching-switch-expressions-and-statements.html.
Looks like this expression one is standard since Java 14. So, feel free to fix it as you wish 😅

Reply:
Thanks for update @artembilan . I was planning on getting to this some time today.